ting_subsearch_suggestions.module

```diff
@@ -179,44 +184,68 @@ function ting_subsearch_suggestions_cron() {
-    $params['end_period'] = variable_get('ting_subsearch_kpi_end_period', NULL);
-    $params['format'] = variable_get('subsearch_kpi_format', NULL);
-    $params['language'] = variable_get('subsearch_kpi_language', NULL);
 
-    $account = variable_get('ting_subsearch_kpi_account', NULL);
-    $password = variable_get('ting_subsearch_kpi_password', NULL);
-
-    if (empty($url) || empty($account) || empty($password)) {
+    if (empty($url)) {
       watchdog('ting_subsearch_opensuggestions', 'KPI: Module not configured properly', array(), WATCHDOG_WARNING);
       return;
     }
 
     try {
-      $client = new HttpClient();
-      $response = $client->get(
-        $url,
-        array(
-          'query' => $params,
-          'headers' => array(
-            'Accept' => 'application/json',
-          ),
-          'auth' => array(
-            $account,
-            $password,
-          ),
-        )
-      );
-      $result = json_decode($response->getBody(), TRUE);
+      // Fetch data from feed.
+      $ch = curl_init($url);
+      curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+      $ret = curl_exec($ch);
+      curl_close($ch);
+
+      $results = str_getcsv($ret, "\r\n");
+      unset($results[0]);
+
+      $prepared = array();
+
+      foreach ($results as $result) {
+        $result = explode(';', trim($result));
+        $prepared[] = array(
+          'weekno' => $result[0],
+          'oss' => $result[1],
+          'results' => $result[2],
+          'oss_r' => $result[3],
+        );
+      }
     }
-    catch (GuzzleClientException $e) {
-      throw new Exception($e->getMessage(), $e->getCode(), $e);
+    catch (Exception $e) {
+      watchdog('cron', $e->getMessage());
     }
 
-    if (!empty($result)) {
-      ting_subsearch_suggestions_insert_records($result);
-      variable_set('subsearch_cron_run', time() + (60 * 60 * 24));
-      watchdog('ting_subsearch_suggestions', 'Data fetched from KPI. Will re-run in a day.', array(), WATCHDOG_INFO);
+    if (!empty($prepared)) {
+      // Check if week is already in db.
+      // If it exists - remove all rows related to this week.
+      $feed_week_nrs = array();
+      foreach ($prepared as $item) {
+        $feed_week_nrs[$item['weekno']] = TRUE;
+      }
+
+      $week_nrs_db = db_select('ting_subsearch_kpi_index_popular_keys', 'tskpi')
+        ->distinct()
+        ->fields('tskpi', array('weekno'))
+        ->execute()
+        ->fetchCol();
+
+      foreach (array_keys($feed_week_nrs) as $feed_week_nr) {
+        if (in_array($feed_week_nr, $week_nrs_db)) {
+          // Remove this week from db.
+          _ting_subsearch_suggestions_clear_passed_week($feed_week_nr);
+        }
+      }
+
+      try {
+        ting_subsearch_suggestions_insert_records($prepared);
+      }
+      catch (Exception $e) {
+        watchdog_exception('ting_subsearch_suggestions', $e->getMessage());
+      }
+
+      variable_set('subsearch_cron_run', time() + (60 * 60 * 24 * 7));
```

Comment:
`REQUEST_TIME` should be used instead of `time()`.
Also `60 * 60 * 24 * 7` can be replaced by a constant denoting needed value.